src/mon/OSDMonitor.cc

```diff
@@ -2024,12 +2024,26 @@ void OSDMonitor::encode_pending(MonitorDBStore::TransactionRef t)
   // metadata, too!
   for (map<int,bufferlist>::iterator p = pending_metadata.begin();
        p != pending_metadata.end();
-       ++p)
+       ++p) {
+    Metadata m;
+    auto mp = p->second.cbegin();
+    decode(m, mp);
+    auto it = m.find("osd_objectstore");
+    if (it != m.end()) {
+        if (it->second == "filestore") {
```

Comment:
Minor nit: Tab for the inner if..else block can be reduced to 2 spaces to be consistent with the rest of the file.